dogfooding/lib/core/repos/custom_environment_loader.dart

```diff
@@ -0,0 +1,29 @@
+/// If you want to use dogfooding with your own environment, you can provide a custom environment loader here.
+/// This will allow you to use your own tokens and API key.
+///
+/// You can use https://getstream.io/chat/docs/flutter-dart/tokens_and_authentication/#manually-generating-tokens to generate tokens and hardcode them.
+///
+/// Replace the global variable `customEnvironmentLoader` with your own implementation.
+///
+/// ```dart
+/// final customEnvironmentLoader = CustomEnvironmentLoader(
+///     tokenLoader: (userId) => '{USER_TOKEN}', apiKey
+///     apiKey: '{API_KEY}');
+/// ```
+class CustomEnvironmentLoader {
+  final String Function(String) tokenLoader;
```

Comment:
Probably better to make this async, so you can download a token from a backend?
```suggestion
  final Future<String> Function(String) tokenLoader;
```

Reply:
replaced with FutureOr 👍